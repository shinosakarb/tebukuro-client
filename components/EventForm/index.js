// @flow
import React, { Component } from 'react'
import TextInput from '../forms/TextInput'

type Props = { onSubmit: Function }
type State = {
  name: string,
  description: string,
  quota: number
}

export default class EventForm extends Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = { name: '', description: '', quota: 0 }
  }

  onChangeHandler = (e: SyntheticInputEvent<>) => {
    this.setState({
      ...this.state,
      [e.target.id]: e.target.value,
    })
  }

  onSubmitHandler = (e: SyntheticEvent<>) => {
    e.preventDefault()
    this.props.onSubmit(this.state)
  }

  render() {
    return (
      <div>
        <h3>Event registration form</h3>
        <form onSubmit={this.onSubmitHandler}>
          <TextInput id="name" value={this.state.name} onChange={this.onChangeHandler} />
          <TextInput id="description" value={this.state.description} onChange={this.onChangeHandler} />
          <TextInput id="quota" value={this.state.quota} onChange={this.onChangeHandler} />
          <input type="submit" />
        </form>
      </div>
    )
  }
}
