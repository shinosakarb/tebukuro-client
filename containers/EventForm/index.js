// @flow
import React from 'react'
import { connect } from 'react-redux'

export const EventForm = () => (
  <div>This is the EventForm Container.</div>
)

export default connect(null)(EventForm)
