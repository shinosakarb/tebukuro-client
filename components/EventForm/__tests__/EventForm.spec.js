import React from 'react'
import { mount, shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import EventForm from '../index'
import EventParams from '../../../factories/Event'

const inputValues = {
  name: EventParams.event1.name,
  description: EventParams.event1.description,
  quota: EventParams.event1.quota,
}

describe('EventForm', () => {
  it('render event form component', () => {
    const wrapper = shallow(<EventForm onSubmit={jest.fn()} />)
    const tree = shallowToJson(wrapper)

    expect(tree).toMatchSnapshot()
  })

  describe('when submit', () => {
    describe('with correct argument', () => {
      const onSubmit = jest.fn()
      const wrapper = mount(<EventForm onSubmit={onSubmit} />)

      const nameElement = wrapper.find('#name')
      const descriptionElement = wrapper.find('#description')
      const quotaElement = wrapper.find('#quota')

      nameElement.simulate(
        'change',
        { target: { id: 'name', value: inputValues.name } },
      )
      descriptionElement.simulate(
        'change',
        { target: { id: 'description', value: inputValues.description } },
      )
      quotaElement.simulate(
        'change',
        { target: { id: 'quota', value: inputValues.quota } },
      )
      wrapper.find('[type="submit"]').simulate('submit')

      it('should call onSubmit once.', () => {
        expect(onSubmit).toHaveBeenCalledTimes(1)
      })

      it('should call onSubmit with correct argument.', () => {
        expect(onSubmit).toBeCalledWith(inputValues)
      })
    })
  })
})
