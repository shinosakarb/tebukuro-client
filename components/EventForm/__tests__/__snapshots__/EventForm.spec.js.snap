// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm render event form component 1`] = `
<div>
  <h3>
    Event registration form
  </h3>
  <form
    onSubmit={[Function]}
  >
    <Component
      id="name"
      onChange={[Function]}
      value=""
    />
    <Component
      id="description"
      onChange={[Function]}
      value=""
    />
    <Component
      id="quota"
      onChange={[Function]}
      value={0}
    />
    <input
      type="submit"
    />
  </form>
</div>
`;
